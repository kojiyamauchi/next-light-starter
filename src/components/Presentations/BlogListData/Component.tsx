/*
  Blog List Data Component.
*/

import styles from './style.module.scss'

type Props = {
  title: string
  date: string
}

export const BlogListData: React.FC<Props> = ({ title, date }): JSX.Element => (
  <div className={styles.wrapper}>
    <h2 className={styles['title-heading']}>{title}</h2>
    <p className={styles['date']}>{date}</p>
  </div>
)
