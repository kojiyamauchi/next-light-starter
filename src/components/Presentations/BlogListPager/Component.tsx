/*
  Blog List Pager Component.
*/

import Link from 'next/link'
import { UrlObject } from 'url'

import { pagesPath } from '@/types/$path'

import styles from './style.module.scss'

type Props = {
  category?: string
  pageIndex: number
  isFirst: boolean
  isLast: boolean
  storybook?: boolean
}

export const BlogListPager: React.FC<Props> = ({ category, pageIndex, isFirst, isLast, storybook }): JSX.Element => {
  const handlePathpida = (index: number | ''): UrlObject => {
    if (category) {
      return pagesPath.blog_category._slug([`${category}`, `${index}`]).$url()
    } else {
      return pagesPath.blog_list._slug([`${index}`]).$url()
    }
  }

  const handleAnchor = {
    prev: (): UrlObject => {
      const index = pageIndex === 2 ? '' : pageIndex - 1
      return handlePathpida(index)
    },
    next: (): UrlObject => {
      const index = pageIndex + 1
      return handlePathpida(index)
    }
  }

  return (
    <ul className={`${styles.wrapper} ${storybook ? styles['is-storybook'] : ''}`}>
      {!isFirst && (
        <li className={styles.prev}>
          <Link href={handleAnchor.prev()}>
            <a rel="prev" className={styles['prev-link']}>
              New.
            </a>
          </Link>
        </li>
      )}
      {!isLast && (
        <li className={styles.next}>
          <Link href={handleAnchor.next()}>
            <a rel="prev" className={styles['next-link']}>
              Old.
            </a>
          </Link>
        </li>
      )}
    </ul>
  )
}
