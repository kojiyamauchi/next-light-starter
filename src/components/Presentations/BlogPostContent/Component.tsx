/*
  Blog Post Content Component.
*/

import styles from './style.module.scss'

type Props = {
  content: string
}

export const BlogPostContent: React.FC<Props> = ({ content }): JSX.Element => (
  <div className={styles.wrapper} dangerouslySetInnerHTML={{ __html: content }}></div>
)
