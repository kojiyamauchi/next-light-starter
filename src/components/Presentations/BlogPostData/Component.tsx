/*
  Blog Post Data Component.
*/

import Link from 'next/link'
import styles from './style.module.scss'

type Props = {
  categories?: string[]
  date: string
}

export const BlogPostData: React.VFC<Props> = ({ categories, date }): JSX.Element => (
  <ul className={styles.wrapper}>
    <li className={styles.category}>
      {categories?.map((info, index) => (
        <Link key={index} href={`/blog-category/${info}/`}>
          <a className={styles['category-link']}>{info}.</a>
        </Link>
      ))}
    </li>
    <li className={styles.date}>{date}</li>
  </ul>
)
