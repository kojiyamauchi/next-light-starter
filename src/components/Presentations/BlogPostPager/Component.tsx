/*
  Blog Post Pager Component.
*/

import Link from 'next/link'

import { pagesPath } from '@/types/$path'

import styles from './style.module.scss'

type Props = {
  prev: string | false
  next: string | false
  storybook?: boolean
}

export const BlogPostPager: React.FC<Props> = ({ prev, next, storybook }): JSX.Element => (
  <ul className={`${styles.wrapper} ${storybook ? styles['is-storybook'] : ''}`}>
    {prev && (
      <li className={styles.prev}>
        <Link href={pagesPath.blog_post._slug(`${prev}`).$url()}>
          <a rel="prev" className={styles['prev-link']}>
            New Post.
          </a>
        </Link>
      </li>
    )}
    {next && (
      <li className={styles.next}>
        <Link href={pagesPath.blog_post._slug(`${next}`).$url()}>
          <a rel="prev" className={styles['next-link']}>
            Old Post.
          </a>
        </Link>
      </li>
    )}
  </ul>
)
