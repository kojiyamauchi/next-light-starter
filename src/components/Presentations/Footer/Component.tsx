/*
  Footer Component.
*/

import { useState, useEffect } from 'react'
import styles from './style.module.scss'

export type Props = {
  foo?: string
}

export const Footer: React.VFC<Props> = (): JSX.Element => {
  const [year, setYear] = useState<number | null>(null)

  useEffect(() => {
    setYear(new Date().getFullYear())
  }, [])

  return (
    <footer className={styles.wrapper}>
      <a href="http://kojiyamauchi.com" target="_blank" rel="noreferrer" className={styles.link}>
        Â© {year} All The Small Things, All Rights Reserved.
      </a>
    </footer>
  )
}
