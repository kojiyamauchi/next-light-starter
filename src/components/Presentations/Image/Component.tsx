/*
  Image Component.
*/

import { useEffect, useState } from 'react'
import { useRouter } from 'next/router'
import styles from './style.module.scss'

type Props = {
  image: string
  extension: 'webp' | 'svg' | 'png' | 'jpg' | 'gif'
  altProps: string
  unitTestPath?: string
}

type OptionalKeys =
  | 'mobileWidth'
  | 'desktopWidth'
  | 'mobileMinWidth'
  | 'desktopMinWidth'
  | 'mobileMaxWidth'
  | 'desktopMaxWidth'
  | 'mobileHeight'
  | 'desktopHeight'
  | 'mobileMinHeight'
  | 'desktopMinHeight'
  | 'mobileMaxHeight'
  | 'desktopMaxHeight'
  | 'margin'
  | 'mobileMargin'
  | 'desktopMargin'

type OptionalProps = {
  [P in OptionalKeys]: string
}

export const Image: React.VFC<Partial<OptionalProps> & Props> = ({ image, extension, altProps, unitTestPath, ...props }): JSX.Element => {
  const mediaMatch = typeof window !== 'undefined' && window.matchMedia('(min-width: 768px)')
  const [firstRender, setFirstRender] = useState(true)
  const [matches, setMatches] = useState(mediaMatch && mediaMatch.matches)
  const router = useRouter()
  const pickBasePath = process.env.envMode ? `${router.basePath}/` : unitTestPath ? unitTestPath : ''

  useEffect((): void | (() => void) => {
    setFirstRender(false)
    const matchesDispatch = (event: MediaQueryListEvent): void => setMatches(event.matches)

    if (mediaMatch) {
      mediaMatch.addEventListener('change', matchesDispatch)
      return (): void => mediaMatch.removeEventListener('change', matchesDispatch)
    }
  }, [mediaMatch])

  const path = extension === 'webp' ? 'webps' : 'images'

  const stylesProps = (matches: boolean): React.CSSProperties => {
    return firstRender
      ? {
          display: 'none'
        }
      : {
          display: 'block',
          width: matches ? props.desktopWidth || '100%' : props.mobileWidth || '100%',
          minWidth: matches ? props.desktopMinWidth || 'none' : props.mobileMinWidth || 'none',
          maxWidth: matches ? props.desktopMaxWidth || 'none' : props.mobileMaxWidth || 'none',
          height: matches ? props.desktopHeight || 'auto' : props.mobileHeight || 'auto',
          minHeight: matches ? props.desktopHeight || 'none' : props.mobileHeight || 'none',
          maxHeight: matches ? props.desktopMaxHeight || 'none' : props.mobileMaxHeight || 'none',
          margin: props.margin ? props.margin : matches ? props.desktopMargin || '0 auto' : props.mobileMargin || '0 auto'
        }
  }

  return (
    <>
      {firstRender ? (
        <>
          <img src={`${pickBasePath}images/loading.svg`} />
        </>
      ) : (
        <img className={styles.image} src={`${pickBasePath}${path}/${image}.${extension}`} alt={altProps} loading="lazy" style={{ ...stylesProps(matches) }} />
      )}
    </>
  )
}
