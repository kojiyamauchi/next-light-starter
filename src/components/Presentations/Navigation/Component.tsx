/*
  Navigation Component.
*/

import Link from 'next/link'

import { pagesPath } from '@/types/$path'

import styles from './style.module.scss'

export const Navigation: React.FC = (): JSX.Element => {
  const link = [
    { path: pagesPath.$url().pathname, letter: 'Primary Page', id: 1 },
    { path: pagesPath.secondary.$url().pathname, letter: 'Secondary Page', id: 2 },
    { path: pagesPath.third.$url().pathname, letter: 'Third Page', id: 3 },
    { path: pagesPath.blog_list._slug().$url().pathname, letter: 'Blog Page', id: 4 }
  ]

  const navigation = (): JSX.Element[] => {
    return link.map(
      (info): JSX.Element => (
        <li key={info.id}>
          <Link href={`${info.path}`}>
            <a className={styles.link}>{info.letter}</a>
          </Link>
        </li>
      )
    )
  }

  return <ul className={styles.wrapper}>{navigation()}</ul>
}
