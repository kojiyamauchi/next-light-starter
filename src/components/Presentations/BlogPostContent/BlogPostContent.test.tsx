/**
 * @jest-environment jsdom
 */
/*
  Blog Post Content Component Unit Test.
*/

import { BlogPostContent } from './Component'
import { render /* ,screen */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
// import userEvent from '@testing-library/user-event'

describe('Blog Post Content Component Unit Test', () => {
  it('Snap Shot Testing', () => {
    const { asFragment } = render(<BlogPostContent content="<div>Snap Shot Test.</div>" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
