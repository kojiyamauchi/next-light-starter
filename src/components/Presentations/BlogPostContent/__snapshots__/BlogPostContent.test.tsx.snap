// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Post Content Component Unit Test Snap Shot Testing 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div>
      Snap Shot Test.
    </div>
  </div>
</DocumentFragment>
`;
